### PR TITLE
API-1835: save the unspecified resources for inspection

### PR DESCRIPTION
**Pull request by deads2k**
This is really useful for debugging purposes and it won't work in production because the exit code will always be non-zero and we have no way of know where to create these

### CODE REVIEW

File: pkg/library/libraryapplyconfiguration/apply_configuration.go [hunk 1]
```diff
@@ -66,7 +55,18 @@ func ValidateAllDesiredMutationsGetter(allDesiredMutationsGetter AllDesiredMutat
 	combinedList.GeneratedNameResources = append(combinedList.GeneratedNameResources, allAllowedOutputResources.UserWorkloadResources.GeneratedNameResources...)
 	filteredMutationRequests := FilterSerializedRequests(allMutationRequests, combinedList)
 
-	if unspecifiedOutputResources := manifestclient.DifferenceOfSerializedRequests(allMutationRequests, filteredMutationRequests); len(unspecifiedOutputResources) > 0 {
+	return manifestclient.DifferenceOfSerializedRequests(allMutationRequests, filteredMutationRequests)
+}
+
+func ValidateAllDesiredMutationsGetter(allDesiredMutationsGetter AllDesiredMutationsGetter, allAllowedOutputResources *libraryoutputresources.OutputResources) error {
+	errs := []error{}
```
Comment (p0lyn0mial) [inline pkg/library/libraryapplyconfiguration/apply_configuration.go:62]: nit: move closer to where it is used.

File: pkg/library/libraryapplyconfiguration/apply_configuration.go [hunk 2]
```diff
@@ -66,7 +55,18 @@ func ValidateAllDesiredMutationsGetter(allDesiredMutationsGetter AllDesiredMutat
 	combinedList.GeneratedNameResources = append(combinedList.GeneratedNameResources, allAllowedOutputResources.UserWorkloadResources.GeneratedNameResources...)
 	filteredMutationRequests := FilterSerializedRequests(allMutationRequests, combinedList)
 
-	if unspecifiedOutputResources := manifestclient.DifferenceOfSerializedRequests(allMutationRequests, filteredMutationRequests); len(unspecifiedOutputResources) > 0 {
+	return manifestclient.DifferenceOfSerializedRequests(allMutationRequests, filteredMutationRequests)
+}
+
+func ValidateAllDesiredMutationsGetter(allDesiredMutationsGetter AllDesiredMutationsGetter, allAllowedOutputResources *libraryoutputresources.OutputResources) error {
```
Comment (p0lyn0mial) [inline pkg/library/libraryapplyconfiguration/apply_configuration.go:61]: maybe this function could also return `unspecifiedOutputResources` ?

**Review by bertinatto — COMMENTED** [review]
I understand we determine whether a resource is unspecified if it's found in allMutationsRequests, but not in allAllowedOutputResources.

/lgtm

Holding in case you want to address Luckasz's comments. Feel free to unhold whenever you guys want.

File: pkg/library/libraryapplyconfiguration/apply_configuration.go (hunk 2 above)
Comment (deads2k) [reply pkg/library/libraryapplyconfiguration/apply_configuration.go:61]: > maybe this function could also return `unspecifiedOutputResources` ?

I prefer my validators only return errors.

File: pkg/library/libraryapplyconfiguration/apply_configuration.go (hunk 1 above)
Comment (deads2k) [reply pkg/library/libraryapplyconfiguration/apply_configuration.go:62]: > nit: move closer to where it is used.

will eventually followup.

**Pull request merged**